lib/spack/spack/test/cmd/style.py

```diff
@@ -92,6 +95,63 @@ def test_changed_files_all_files(flake8_package):
     assert not any(f.startswith(spack.paths.external_path) for f in files)
 
 
+@pytest.mark.skipif(not which('git'), reason='requires git')
+def test_external_root(flake8_package_with_errors, tmpdir):
+    """Ensure we can run in a separate root directory w/o configuration files."""
+    git = which("git", required=True)
+    with tmpdir.as_cwd():
+        git
+
+    # create a sort-of spack-looking directory
+    script = tmpdir / "bin" / "spack"
+    script.ensure()
+    spack_dir = tmpdir / "lib" / "spack" / "spack"
+    spack_dir.ensure("__init__.py")
+    llnl_dir = tmpdir / "lib" / "spack" / "llnl"
+    llnl_dir.ensure("__init__.py")
+
+    # create a base develop branch
+    with tmpdir.as_cwd():
+        git("init")
+        git("add", ".")
+        git("commit", "-m", "initial commit")
+        git("branch", "-m", "develop")
+        git("checkout", "-b", "feature")
```

Comment:
What about the case where the name of the branch is already develop?

Reply:
I don't think we need to check it because `develop...` will just evaluate to whatever `develop` is. I don't think it gets covers any new code paths.